servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContextProvider.java

```diff
@@ -40,200 +41,214 @@
     <T> PublisherSource.Subscriber<T> wrapPublisherSubscriberAndSubscription(PublisherSource.Subscriber<T> subscriber,
-                                                                             AsyncContextMap current);
+                                                                             ContextMap current);
 
     /**
      * Wrap an {@link Executor} to ensure it is able to track {@link AsyncContext} correctly.
+     *
      * @param executor The executor to wrap.
      * @return The wrapped executor.
      */
     java.util.concurrent.Executor wrapJdkExecutor(java.util.concurrent.Executor executor);
 
     /**
      * Wrap an {@link ExecutorService} to ensure it is able to track {@link AsyncContext} correctly.
+     *
      * @param executor The executor to wrap.
      * @return The wrapped executor.
      */
     ExecutorService wrapJdkExecutorService(ExecutorService executor);
 
     /**
-     * Wrap an {@link Executor} to ensure it is able to track {@link AsyncContext}
-     * correctly.
+     * Wrap an {@link Executor} to ensure it is able to track {@link AsyncContext} correctly.
+     *
      * @param executor The executor to wrap.
      * @return The wrapped executor.
      */
     Executor wrapExecutor(Executor executor);
 
-    /**
-     * Wrap a {@link CompletableFuture} so that {@link AsyncContext} is preserved from listener methods.
-     * @param future The future to wrap.
-     * @param <T> The type of data for {@link CompletableFuture}.
-     * @return the wrapped {@link CompletableFuture}.
-     */
-    <T> CompletableFuture<T> wrapCompletableFuture(CompletableFuture<T> future, AsyncContextMap current);
-
     /**
      * Wrap a {@link ScheduledExecutorService} to ensure it is able to track {@link AsyncContext} correctly.
      * @param executor The executor to wrap.
      * @return The wrapped executor.
      */
     ScheduledExecutorService wrapJdkScheduledExecutorService(ScheduledExecutorService executor);
 
+    /**
+     * Wrap a {@link CompletableFuture} so that {@link AsyncContext} is preserved from listener methods.
+     *
+     * @param future The future to wrap.
+     * @param current The current {@link ContextMap}.
+     * @param <T> The type of data for {@link CompletableFuture}.
+     * @return the wrapped {@link CompletableFuture}.
+     */
+    <T> CompletableFuture<T> wrapCompletableFuture(CompletableFuture<T> future, ContextMap current);
+
     /**
      * Wrap a {@link Runnable} to ensure it is able to track {@link AsyncContext} correctly.
+     *
      * @param runnable The runnable to wrap.
-     * @param contextMap The {@link AsyncContext}.
+     * @param current The current {@link ContextMap}.
      * @return The wrapped {@link Runnable}.
      */
-    Runnable wrapRunnable(Runnable runnable, AsyncContextMap contextMap);
+    Runnable wrapRunnable(Runnable runnable, ContextMap current);
 
     /**
      * Wrap a {@link Callable} to ensure it is able to track {@link AsyncContext} correctly.
```

Comment:
Perhaps change to `ContextMap` as well?

Reply:
This is a helper pkg-private interface to support `AsyncContext`, referencing `AsyncContext` from here is intentional.